lib/opentelemetry/src/convert.rs

```diff
@@ -0,0 +1,133 @@
+        } else {
+            self.instrumentation_library_logs
+                .into_iter()
+                .map(ScopeLogs::from)
+                .collect()
+        };
+
+        scope_logs
+            .into_iter()
+            .flat_map(|scope_log| scope_log.log_records)
+            .map(|log_record| {
+                ResourceLog {
+                    resource: resource.clone(),
+                    log_record,
+                }
+                .into()
+            })
+            .collect::<Vec<Event>>()
+            .into_iter()
+    }
+}
+
+impl From<PBValue> for Value {
+    fn from(av: PBValue) -> Self {
+        match av {
+            PBValue::StringValue(v) => Value::Bytes(Bytes::from(v)),
+            PBValue::BoolValue(v) => Value::Boolean(v),
+            PBValue::IntValue(v) => Value::Integer(v),
+            PBValue::DoubleValue(v) => Value::Float(NotNan::new(v).unwrap()),
+            PBValue::BytesValue(v) => Value::Bytes(Bytes::from(v)),
+            PBValue::ArrayValue(arr) => Value::Array(
+                arr.values
+                    .into_iter()
+                    .map(|av| av.value.map(Into::into).unwrap_or(Value::Null))
+                    .collect::<Vec<Value>>(),
+            ),
+            PBValue::KvlistValue(arr) => kv_list_into_value(arr.values),
+        }
+    }
+}
+
+struct ResourceLog {
+    resource: Option<OtelResource>,
+    log_record: LogRecord,
+}
+
+fn kv_list_into_value(arr: Vec<KeyValue>) -> Value {
+    Value::Object(
+        arr.into_iter()
+            .filter_map(|kv| {
+                kv.value
+                    .map(|av| (kv.key, av.value.map(Into::into).unwrap_or(Value::Null)))
+            })
+            .collect::<BTreeMap<String, Value>>(),
+    )
+}
+
+impl From<ResourceLog> for Event {
+    fn from(rl: ResourceLog) -> Self {
+        let mut le = LogEvent::default();
+        if let Some(resource) = rl.resource {
+            le.insert(RESOURCE_KEY, kv_list_into_value(resource.attributes));
+        }
+        if !rl.log_record.attributes.is_empty() {
+            le.insert(ATTRIBUTES_KEY, kv_list_into_value(rl.log_record.attributes));
+        }
+        if let Some(v) = rl.log_record.body.and_then(|av| av.value) {
+            le.insert(log_schema().message_key(), v);
+        }
+        le.insert(
+            log_schema().timestamp_key(),
+            rl.log_record.time_unix_nano as i64,
+        );
+        if !rl.log_record.trace_id.is_empty() {
+            le.insert(
+                TRACE_ID_KEY,
+                Value::Bytes(Bytes::from(hex::encode(rl.log_record.trace_id))),
+            );
+        }
+        if !rl.log_record.span_id.is_empty() {
+            le.insert(
+                SPAN_ID_KEY,
+                Value::Bytes(Bytes::from(hex::encode(rl.log_record.span_id))),
+            );
+        }
+        le.insert(SEVERITY_TEXT_KEY, rl.log_record.severity_text);
+        le.insert(SEVERITY_NUMBER_KEY, rl.log_record.severity_number as i64);
```

Comment:
Severity and severity number are also optional. Can we suppress inserting if they aren't specified (0 or empty string).